Code looks ok to me.  
However when I tried to build on Jenkins it fails:
14:26:35 ⚡ Done in 188ms
14:26:35 [ 36%] Built target portal-js
14:26:35 [ 37%] Generating ../PortalHTML/ConfiguringHPCC_EN_US/eclipsehelp.css, ../PortalHTML/ConfiguringHPCC_EN_US/index.js
14:26:35 cp: cannot stat '/jenkins/workspace/DocBuild-JD02/HPCC-Platform/build/docs/portal-js/dist/index.js': No such file or directory
14:26:35 gmake[2]: *** [docs/EN_US/ConfiguringHPCC/CMakeFiles/configuringHPCC_html_EN_US.dir/build.make:76: docs/EN_US/PortalHTML/ConfiguringHPCC_EN_US/eclipsehelp.css] Error 1
14:26:35 gmake[2]: *** Deleting file 'docs/EN_US/PortalHTML/ConfiguringHPCC_EN_US/eclipsehelp.css'
14:26:35 gmake[1]: *** [CMakeFiles/Makefile2:5904: docs/EN_US/ConfiguringHPCC/CMakeFiles/configuringHPCC_html_EN_US.dir/all] Error 2
14:26:35 gmake: *** [Makefile:166: all] Error 2
14:26:35 Build step 'CMake Build' marked build as failure
14:26:35 Archiving artifacts
14:26:37 Finished: FAILURE
 
looks like failing on index.js  

could it be missing something form a cmake.txt file ?

--- a/cmake_modules/docMacros.cmake
+++ b/cmake_modules/docMacros.cmake
@@ -110,6 +110,9 @@ MACRO(DOCBOOK_TO_HTML _xsl_file _xml_file _out_dir _html_target _css_path _zip_t
        endif()
        get_filename_component(css_file_name ${_css_path} NAME)
 
+       set(_portaljs_path "${CMAKE_BINARY_DIR}/docs/portal-js/dist/index.js")
+       get_filename_component(portaljs_file_name ${_portaljs_path} NAME)
+
        STRING(REGEX REPLACE "(.+)/([^/]+)$" "\\1" _out_dir1 "${_out_dir}")
        STRING(REGEX REPLACE ".+/([^/]+)$" "\\1" _out_dir2 "${_out_dir}")
 
@@ -119,8 +122,10 @@ MACRO(DOCBOOK_TO_HTML _xsl_file _xml_file _out_dir _html_target _css_path _zip_t
        ADD_CUSTOM_COMMAND(
            COMMAND mkdir -p ${_out_dir}
            COMMAND cp ${_css_path} ${_out_dir}/
-           OUTPUT ${_out_dir}/${css_file_name}
-           )
+           COMMAND cp ${_portaljs_path} ${_out_dir}/
+           OUTPUT ${_out_dir}/${css_file_name} ${_out_dir}/${portaljs_file_name}
+           DEPENDS portal-js
+       )
        ADD_CUSTOM_TARGET(${_html_target}
            COMMAND ${XSLTPROC_EXECUTABLE} --nonet --xinclude --stringparam html.stylesheet ${css_file_name} --stringparam generate.toc "book toc"  --param chapter.autolabel 0  ${_xsl_file} ${_xml_file}
            WORKING_DIRECTORY ${_out_dir}

--- a/docs/BuildTools/PortalGen.xsl
+++ b/docs/BuildTools/PortalGen.xsl
@@ -34,4 +34,11 @@
    <BR/>
  </xsl:template>
 
+<!-- Colourize ECL Code -->
+  <xsl:template name="user.head.content">
+  </xsl:template>
+  <xsl:template name="user.footer.content">
+      <script src="index.js" />
+  </xsl:template>
+
 </xsl:stylesheet>

--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -55,6 +55,7 @@ endif()
 
 
 # Required directories to add.
+add_subdirectory(portal-js)
 add_subdirectory(common)
 add_subdirectory(BuildTools)
 add_subdirectory(resources)

--- a/docs/EN_US/ECLProgrammersGuide/PRG_Mods/CodeSign.xml
+++ b/docs/EN_US/ECLProgrammersGuide/PRG_Mods/CodeSign.xml
@@ -146,7 +146,7 @@
     Server machine, so if you want to use your own service definitions, they
     can be signed using a key that has been installed in this way:</para>
 
-    <para><programlisting>gpg --output &lt;signed-ecl&gt; --default-key &lt;key-id&gt; --clearsign &lt;ecl-file-to-sign&gt;</programlisting></para>
+    <para><programlisting lang="bash">gpg --output &lt;signed-ecl&gt; --default-key &lt;key-id&gt; --clearsign &lt;ecl-file-to-sign&gt;</programlisting></para>
 
     <para>Using this method, a trusted person can sign code to indicate that
     it is acceptable for untrusted people to use, without allowing the

--- a/docs/portal-js/CMakeLists.txt
+++ b/docs/portal-js/CMakeLists.txt
@@ -1,0 +1,82 @@
+project ( portal-js ) 
+
+file(REMOVE_RECURSE
+    ${CMAKE_CURRENT_BINARY_DIR}/src
+)
+
+file(COPY
+    ${CMAKE_CURRENT_SOURCE_DIR}/src
+    ${CMAKE_CURRENT_SOURCE_DIR}/package-lock.json
+    ${CMAKE_CURRENT_SOURCE_DIR}/package.json
+    DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+
+set ( SRCS
+    ${CMAKE_CURRENT_BINARY_DIR}/src/index.ts
+    ${CMAKE_CURRENT_SOURCE_DIR}/package-lock.json
+    ${CMAKE_CURRENT_SOURCE_DIR}/package.json
+)
+
+if ( "${CMAKE_BUILD_TYPE}" STREQUAL "Debug" )
+    set ( PORTALJS_BUILD "build-dev" )
+else ()
+    set ( PORTALJS_BUILD "build" )
+endif ()
+
+set (NPM_ERROR "")
+if (WIN32)
+    execute_process(COMMAND cmd /c "npm --version" OUTPUT_VARIABLE _VERSION RESULT_VARIABLE _NPM_VERSION_RESULT)
+else ()
+    execute_process(COMMAND npm --version OUTPUT_VARIABLE _VERSION RESULT_VARIABLE _NPM_VERSION_RESULT)
+endif ()
+if (NOT _NPM_VERSION_RESULT)
+    string (REPLACE "v" "" NPM_VERSION_STRING "${_VERSION}")
+    string (REPLACE "." ";" _VERSION_LIST "${NPM_VERSION_STRING}")
+    list (GET _VERSION_LIST 0 NPM_VERSION_MAJOR)
+    if (NPM_VERSION_MAJOR LESS 6)
+        set (NPM_ERROR "NPM version ${NPM_VERSION_MAJOR} is too old (expected >= 6), please install NodeJS as per https://github.com/hpcc-systems/HPCC-Platform/wiki/Building-HPCC#prerequisites" )
+    endif ()
+else ()
+    set (NPM_ERROR "Unable to locate node/npm, please install NodeJS as per https://github.com/hpcc-systems/HPCC-Platform/wiki/Building-HPCC#prerequisites" )
+endif ()
+
+if ( NOT "${NPM_ERROR}" STREQUAL "")
+    message ( FATAL_ERROR "${NPM_ERROR}" )
+endif ()
+
+add_custom_command (
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/no.file
+    COMMAND npm run --silent clean
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+)
+
+add_custom_target( portaljs_clean
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/no.file
+)
+
+add_custom_command (
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/node_modules/.package-lock.json
+    COMMAND npm ci 
+    # --silent
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/package-lock.json
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+)
+
+add_custom_target( portaljs_fetchdeps
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/node_modules/.package-lock.json
+)
+
+add_custom_command (
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/dist/index.js
+    COMMAND npm run --silent ${PORTALJS_BUILD}
+    DEPENDS ${SRCS}
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/node_modules/.package-lock.json
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+)
+
+add_custom_target ( portal-js ALL 
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/dist/index.js
+)
+
+add_dependencies ( portal-js portaljs_fetchdeps )
+
+# install ( DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/build/." DESTINATION componentfiles/files COMPONENT Runtime USE_SOURCE_PERMISSIONS )

--- a/docs/portal-js/package.json
+++ b/docs/portal-js/package.json
@@ -1,0 +1,23 @@
+{
+  "name": "portal-js",
+  "private": true,
+  "version": "1.0.0",
+  "description": "",
+  "type": "module",
+  "main": "index.js",
+  "scripts": {
+    "clean": "rimraf ./dist",
+    "build-dev": "esbuild ./src/index.ts --format=iife --bundle --outfile=./dist/index.js",
+    "build": "esbuild ./src/index.ts --format=iife --bundle --minify --outfile=./dist/index.js",
+    "watch": "esbuild ./src/index.ts --format=iife --bundle --watch --outfile=./dist/index.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@hpcc-js/common": "2.71.9",
+    "@hpcc-js/codemirror": "2.60.9",
+    "esbuild": "0.18.14",
+    "rimraf": "5.0.1"
+  }
+}

--- a/docs/portal-js/src/index.ts
+++ b/docs/portal-js/src/index.ts
@@ -1,0 +1,36 @@
+import { select } from "@hpcc-js/common";
+import { ECLEditor } from "@hpcc-js/codemirror";
+
+select("body").selectAll("pre.programlisting:not([lang]),pre.programlisting[lang='ecl']")
+    .each(function (d, i) {
+        if (this) {
+            const element = this as HTMLPreElement;
+            const bbox = element.getBoundingClientRect();
+
+            const div = document.createElement("div");
+            div.style.marginLeft = "3.25em";
+            div.style.marginTop = "8px";
+            div.style.marginBottom = "8px";
+            if (element.parentNode) {
+                element.parentNode.insertBefore(div, element);
+
+                const d3Element = select(this);
+
+                const cm = new ECLEditor()
+                    .target(div)
+                    .resize({ width: Math.max(800, bbox.width), height: bbox.height })
+                    .text(d3Element.text().trim())
+                    .readOnly(true)
+                    .render()
+                    ;
+
+                d3Element.remove();
+
+                window.addEventListener("resize", function () {
+                    cm.resize().lazyRender();
+                });
+
+            }
+        }
+    })
+    ;

